src/test/java/seedu/address/logic/commands/ViewCommandTest.java

```diff
@@ -0,0 +1,92 @@
+import seedu.address.model.patient.IcNumber;
+import seedu.address.model.patient.PatientWithIcNumberPredicate;
+import seedu.address.model.patient.exceptions.PatientWithFieldNotFoundException;
+
+
+/**
+ * Contains integration tests (interaction with the Model) for {@code ViewCommand}.
+ */
+public class ViewCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    private Model expectedModel = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+
+    @Test
+    public void equals() throws PatientWithFieldNotFoundException {
+        IcNumber testIcNumber1 = new IcNumber("t1234567a");
+        IcNumber testIcNumber2 = new IcNumber("s7654321a");
+
+        PatientWithIcNumberPredicate firstPredicate = new PatientWithIcNumberPredicate(testIcNumber1);
+        PatientWithIcNumberPredicate secondPredicate = new PatientWithIcNumberPredicate(testIcNumber2);
+
+        ViewCommand viewFirstCommand = new ViewCommand(firstPredicate, testIcNumber1);
+        ViewCommand viewSecondCommand = new ViewCommand(secondPredicate, testIcNumber2);
+
+        // same object -> returns true
+        assertTrue(viewFirstCommand.equals(viewFirstCommand));
+
+        // same values -> returns true
+        ViewCommand viewFirstCommandCopy = new ViewCommand(firstPredicate, testIcNumber1);
+        assertTrue(viewFirstCommand.equals(viewFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(viewFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(viewFirstCommand.equals(null));
+
+        // different patient -> returns false
+        assertFalse(viewFirstCommand.equals(viewSecondCommand));
+    }
+
+    @Test
+    public void execute_icNumberOfExistingPatient_patientFound() throws PatientWithFieldNotFoundException {
+        IcNumber testIcNumber1 = new IcNumber("T0032415E");
+        PatientWithIcNumberPredicate predicate = new PatientWithIcNumberPredicate(testIcNumber1);
+        ViewCommand command = new ViewCommand(predicate, testIcNumber1);
+        expectedModel.updateFilteredPatientList(predicate);
+        assertCommandSuccess(command, model, MESSAGE_PATIENT_LISTED_SUCCESS, expectedModel);
+    }
+
+    @Test
+    public void execute_icNumberOfNonExistingPatient_exceptionThrown() throws PatientWithFieldNotFoundException {
+        IcNumber testIcNumber1 = new IcNumber("T1234567j");
+        PatientWithIcNumberPredicate predicate = new PatientWithIcNumberPredicate(testIcNumber1);
+        ViewCommand command = new ViewCommand(predicate, testIcNumber1);
+
+        boolean exceptionThrown = false;
+        try {
+            command.execute(model);
+        } catch (PatientWithFieldNotFoundException e) {
+            exceptionThrown = true;
+            assertEquals(e.getMessage(),
+                MESSAGE_UNABLE_TO_FIND_PATIENT_WITH_FIELD + "Ic Number : " + testIcNumber1.value);
+        }
+        assertTrue(exceptionThrown);
```

Comment:
If I'm not mistaken, you can also use `assertThrows`!

Reply:
got it!